dts/bindings/flash_controller/zephyr,disk-flash.yaml

```diff
@@ -0,0 +1,51 @@
+# SPDX-License-Identifier: Apache-2.0
+
+description: Emulates a flash using a disk
+
+compatible: "zephyr,disk-flash"
+
+include: base.yaml
+
+properties:
+  size:
+    type: int
+    description: flash capacity in bits
+    required: true
+  disk-name:
+    type: string
+    description: the name of the underlying disk to be used
+    required: true
+  disk-sector-size:
+    type: int
+    description: the sector size in byte of the underlying disk to be used
+    required: true
+  disk-offset:
+    type: int
+    description: >
+      offset on the disk in bytes. If the offset is 10 the 10th byte will
+      be the first one used.
+    default: 0
+  init-level:
+    type: string
+    description: >
+      The initialization level of this driver. It is important, that the underlying disk has
+      already been initialized.
+    default: "APPLICATION"
+  init-priority:
```

Comment:
Cannot be devicetree property.

Reply:
Why? I understand that with the new ramdisk cleanup it is now initialized post kernel but should the user not have the freedom to adjust this to his needs?

Reply:
Because it is always a Kconfig option, here for flash drivers FLASH_INIT_PRIORITY.

Reply:
I replaced this with the `FLASH_INIT_PRIORITY`.
-> I still think this is less flexible but it works. It has to be overwritten though since `ramdisk` has the same priority